packages/engine/lib/execution/src/package/state/behavior_execution/behavior/field.rs

```diff
@@ -0,0 +1,99 @@
+use std::collections::HashMap;
+
+use stateful::field::{FieldSpecMap, RootFieldSpec, RootFieldSpecCreator};
+
+use crate::{
+    package::{
+        state::behavior_execution::{behavior::keys::BehaviorKeys, Behavior},
+        PackageInitConfig,
+    },
+    Error, Result,
+};
+
+#[derive(Clone)]
+pub struct BehaviorFieldMap {
```

Comment:
I don't have a good suggestion, but can you think of a better name for this? It's just a bit confusing when it **isn't** a `Map` and we literally have `BehaviorMap` below

Reply:
I had issues finding a name for this, too. I also find the term `Map` confusing but wasn't able to come up with a proper name.

Reply:
Maybe BehaviorAndFields? BehaviorWithFields?

Although we should probably just rename BehaviorKeys to BehaviorFields so maybe add a todo for that and leave the naming as is